### ISSUE TITLE
this will have problem (   obj->shdr = obj->symtab = obj->strtab = NULL;)

### ISSUE THREAD

**Issue by zylinux** [ushitora-anqou/aqcc#16]
looks like forget initial   obj->rela_text to NULL.

https://github.com/ushitora-anqou/aqcc/blob/master/ld/link.c

 obj->shdr = obj->symtab = obj->strtab = NULL;         ->                obj->shdr = obj->symtab = obj->strtab = obj->rela_text =NULL;

        if (obj->rela_text == NULL && strcmp(name, ".rela.text") == 0) {
            obj->rela_text = offset;
            obj->nrela_text = size / 24;
        }

**Comment by ushitora-anqou** [comment]
Yeah, we need to initialize `obj->rela_text` here. Thanks for pointing that out. I don't have time to fix it now, but a PR is welcome.
